spec/lib/jobs/generate-tag-spec.js

```diff
@@ -0,0 +1,256 @@
+            createdAt: new Date('Feb 01 2015 12:00:00'),
+            name: 'Test Tag',
+            rules: [
+                {
+                    path: 'dmi.dmi.system.manufacturer',
+                    regex: /[On]*[Rr]ack[HD]*/
+                }
+            ]
+        },
+        {
+            id: '2',
+            createdAt: new Date('Feb 01 2015 12:00:00'),
+            name: 'Test Tag 2',
+            rules: [
+                {
+                    path: 'dmi.dmi.system.manufacturer',
+                    equals: 'Other'
+                }
+            ]
+        }];
+
+        waterline.tags.find.resolves(tags);
+        waterline.catalogs.findMostRecent.resolves(catalog1);
+
+        return job.run()
+        .then(function() {
+            expect(waterline.nodes.addTags).to.have.been.calledOnce;
+            expect(waterline.nodes.addTags).to.have.been.calledWith(['Test Tag']);
+        });
+    });
+
+    it('assigns nothing when no tags match', function() {
+        var job = new GenerateTag({}, { target: 'bc7dab7e8fb7d6abf8e7d6ab' }, uuid.v4());
+        var tag = {
+            id: '1',
+            createdAt: new Date('Feb 01 2015 12:00:00'),
+            name: 'Test Tag',
+            rules: [
+                {
+                    path: 'dmi.dmi.system.manufacturer',
+                    equals: 'Other'
+                }
+            ]
+        };
+        waterline.tags.find.resolves([tag]);
+        waterline.catalogs.findMostRecent.resolves(catalog1);
+
+        return job.run()
+        .then(function() {
+            expect(waterline.nodes.addTags).to.have.not.been.called;
+        });
+    });
+
+    it('assigns all tags when multiples match', function() {
+        var job = new GenerateTag({}, { target: 'bc7dab7e8fb7d6abf8e7d6ab' }, uuid.v4());
+        var tags = [{
+            id: '1',
+            createdAt: new Date('Feb 01 2015 12:00:00'),
+            name: 'Test Tag',
+            rules: [
+                {
+                    path: 'dmi.dmi.system.manufacturer',
+                    equals: 'RackHD'
+                }
+            ]
+        }, {
+            id: '2',
+            createdAt: new Date('Feb 01 2015 12:00:00'),
+            name: 'Test Tag 2',
+            rules: [
+                {
+                    path: 'dmi.dmi.system.manufacturer',
+                    equals: 'RackHD'
+                },
+                {
+                    path: 'dmi.dmi.system.product_name',
+                    equals: '1A'
+                }
+            ]
+        }];
+        waterline.tags.find.resolves(tags);
+        waterline.catalogs.findMostRecent.resolves(catalog1);
+
+        return job.run()
+        .then(function() {
+            expect(waterline.nodes.addTags).to.have.been.calledOnce;
+            expect(waterline.nodes.addTags).to.have.been.calledWith(['Test Tag', 'Test Tag 2']);
+        });
+    });
+
+    it('assigns a matching tag against multiple catalogs', function() {
```

Comment:
just verifying, this is effectively "assign the tag if both these rules match" - and *`AND`* logical setup? Is there an *`OR`*? mechanism